src/SFA.DAS.Events.Api.UnitTests/Orchestrators/ApprenticeshipEventsOrchestratorTests/WhenICreateAListOfEvents.cs

```diff
@@ -0,0 +1,149 @@
+
+            EventsLogger.Verify(x => x.Warn(validationException, "Invalid apprenticeship event bulk upload request", null, null, null));
+        }
+
+        [Test]
+        public void AndAnExceptionOccursThenTheErrorIsLogged()
+        {
+            var exception = new Exception("Exception");
+            Mediator.Setup(m => m.SendAsync(It.IsAny<BulkUploadCreateApprenticeshipEventsCommand>())).ThrowsAsync(exception);
+
+            Assert.ThrowsAsync<Exception>(() => Orchestrator.CreateEvents(_events));
+
+            EventsLogger.Verify(x => x.Error(exception, exception.Message, null, null, null));
+        }
+
+        private bool BulkCommandMatchesEvents(BulkUploadCreateApprenticeshipEventsCommand command, List<ApprenticeshipEvent> events)
+        {
+            if (command.ApprenticeshipEvents?.Count != events?.Count)
+            {
+                return false;
+            }
+
+            var i = 0;
+            foreach (var domainEvent in command.ApprenticeshipEvents)
+            {
+                if (!DomainEventMatchesApiEvent(domainEvent, events[i]))
+                {
+                    return false;
+                }
+                i++;
+            }
+            return true;
+        }
+
+        private bool DomainEventMatchesApiEvent(Domain.Entities.ApprenticeshipEvent domainEvent,
+            ApprenticeshipEvent apiEvent)
+        {
+
+            return domainEvent.ProviderId == apiEvent.ProviderId &&
+                   domainEvent.AgreementStatus.ToString() == apiEvent.AgreementStatus.ToString() &&
+                   domainEvent.ApprenticeshipId == apiEvent.ApprenticeshipId &&
+                   domainEvent.EmployerAccountId == apiEvent.EmployerAccountId &&
+                   domainEvent.Event == apiEvent.Event &&
+                   domainEvent.LearnerId == apiEvent.LearnerId &&
+                   domainEvent.PaymentOrder == apiEvent.PaymentOrder &&
+                   domainEvent.PaymentStatus.ToString() == apiEvent.PaymentStatus.ToString() &&
+                   domainEvent.PausedOnDate == apiEvent.PausedOnDate &&
+                   domainEvent.StoppedOnDate == apiEvent.StoppedOnDate &&
+                   domainEvent.TrainingEndDate == apiEvent.TrainingEndDate &&
+                   domainEvent.TrainingId == apiEvent.TrainingId &&
+                   domainEvent.TrainingStartDate == apiEvent.TrainingStartDate &&
+                   domainEvent.TrainingTotalCost == apiEvent.TrainingTotalCost &&
+                   domainEvent.TrainingType.ToString() == apiEvent.TrainingType.ToString() &&
+                   domainEvent.LegalEntityId == apiEvent.LegalEntityId &&
+                   domainEvent.LegalEntityName == apiEvent.LegalEntityName &&
+                   domainEvent.LegalEntityOrganisationType == apiEvent.LegalEntityOrganisationType &&
+                   domainEvent.DateOfBirth == apiEvent.DateOfBirth &&
```

Comment:
When we merge this PR with the agreement id PR (https://github.com/SkillsFundingAgency/das-events/pull/64/files), we should add AccountLegalEntityPublicHashedId to this test.